### DISCUSSION TITLE
Help configureing my css, diffrent clor for tree collaps and so on.

### DISCUSSION THREAD

**Discussion by haShinui** [piroor/treestyletab#3342] [Q&A]
Hello, so I found an awesome TST theme and I want to configure it more to more fit me. I went through the code snippets and could find to much.
So what I want is that 1. I dont want that the tree collapses when I drag tabs around.
2. I have a my pinned on the top of my tst but its on the left side, I want they start from the right to the left so that even when I collapses I can see my pinned tabs.
3. Right now I have some color coding, that when a tab is a child there is a purple line on the right, but I want that when I have another level, then its a different color. 
4. That when I have child, it does and indent (so tab is placed a bit to the right) under the parent tab. I have that but just for 1 level, if I have a child of a child of a parent, the 2nd level child parent is like a sibling, so no indent.
I also attached a video where I show everything

https://github.com/piroor/treestyletab/assets/67230963/7ed326f0-94f8-4276-817d-05bf4a286b2e

Also the wierd bug with the pinning of tabs,  I have no Idea why that happend.

**Comment by haShinui** [comment]
Also is there a shortcut to open a tab as a child or as sibling next to current tab?

**Reply by irvinm** [reply]
> Also is there a shortcut to open a tab as a child or as sibling next to current tab?

user_pref("browser.tabs.insertAfterCurrent", true);

**Comment by irvinm** [comment]
> Hello, so I found an awesome TST theme and I want to configure it more to more fit me. I went through the code snippets and could find to much. So what I want is that 1. I dont want that the tree collapses when I drag tabs around. 2. I have a my pinned on the top of my tst but its on the left side, I want they start from the right to the left so that even when I collapses I can see my pinned tabs. 3. Right now I have some color coding, that when a tab is a child there is a purple line on the right, but I want that when I have another level, then its a different color. 4. That when I have child, it does and indent (so tab is placed a bit to the right) under the parent tab. I have that but just for 1 level, if I have a child of a child of a parent, the 2nd level child parent is like a sibling, so no indent. I also attached a video where I show everything
> 2023-05-20.12-56-28.mp4
> 
> Also the wierd bug with the pinning of tabs, I have no Idea why that happend.

There is a lot going on here ... not sure how people are going to help, especially without seeing the CSS you are using for this theme.

**Reply by haShinui** [reply]
So this is the code that makes an indent line, like the pulgin uses the tab-item and tab-item-substance left border and then straightens it out and thickenss it. `tab-item:not([data-level="0"]):not(pinned) {
border-left: solid 3px purple !important;
border-top-left-radius: 0;
border-bottom-left-radius: 0;
margin-left: 20px /* that makes the indent */;
padding-left: 0px;
`
U can modifie that at diffrent data levels there are is a diffrent color, but you can have max 2 indent lines, becuase you just have 1 tab-item and 1 tab-item-substance.
I have more on the issue of this github.

**Reply by irvinm** [reply]
Correct, those are all true statements.

Have you looked over this?  https://github.com/piroor/treestyletab/issues/2948

**Reply by irvinm** [reply]
If you look into the CSS from the helper addon passed into TST, additional SPAN elements are inserted into the tab DOM in order to provide a way to allow additional elements that can have lines\coloring changed.

**Reply by haShinui** [reply]
Hello so I looked into https://github.com/piroor/treestyletab/issues/2948 and  it was helpful, i did not see it before. I also just started to use css with costumizing my firefox, i tried customizing the indent line via the css. But I did not work, I tried calling the function and then overriding it with `!important` but its stayed the same. I am just no sure I exaclty I change indent line, I used [this](https://github.com/piroor/tst-indent-line/blob/main/background/background.js) for help
